connect-inject/handler.go

```diff
@@ -682,16 +682,28 @@ func (h *Handler) shouldRunMergedMetricsServer(pod *corev1.Pod) (bool, error) {
 // determineAndValidatePort behaves as follows:
```

Comment:
I tried to clarify that as:
```go
// serviceMetricsPort returns the port the service exposes metrics on. This will
// default to the port used to register the service with Consul, and can be
// overridden with the annotation if provided.
```

Reply:
👍